### PR TITLE
fix: use correct OS check for settings_build when calling b2 or bcp exe

### PR DESCRIPTION
**Pull request by Pro**
Specify library name and version:  **boost/1.81.0**

The check `self._settings_build == "Windows"` will always return `False`, since it is the settings object.

Correct check is `self._settings_build.os == "Windows"`.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
> The check `self._settings_build == "Windows"` will always return `False`, since it is the settings object.
> 
> Correct check is `self._settings_build.os == "Windows"`.

Therefore it's a good indication that this logic is useless and should be removed, since boost build on Windows didn't fail.

File: recipes/boost/all/conanfile.py [hunk 1]
```diff
@@ -869,7 +869,7 @@ def build(self):
         with chdir(self, sources):
             # To show the libraries *1
             # self.run("%s --show-libraries" % b2_exe)
-            self.run(full_command)
+            self.run(full_command, run_environment=True)
```
Comment (SpaceIm) [inline recipes/boost/all/conanfile.py:872]:
```suggestion
            self.run(full_command)
```

File: recipes/boost/all/conanfile.py [hunk 2]
```diff
@@ -774,12 +774,12 @@ def _clean(self):
 
     @property
     def _b2_exe(self):
-        return "b2.exe" if self._settings_build == "Windows" else "b2"
+        return "b2.exe" if self._settings_build.os == "Windows" else "b2"
```
Comment (SpaceIm) [inline recipes/boost/all/conanfile.py:777]:
```suggestion
        return "b2"
```

**Comment by SpaceIm** [comment]
You can remove `_settings_build` property because it's not used anymore.

**Review by prince-chrismc — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**